ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
So, this version isn't actually safe, I think. If you call `ssl_get_new_session` and leave `s->hit` at 1, you'll resume an empty master secret and be trivially MITM-able.